### DISCUSSION TITLE
All windows moved from monitor to Macbook display after unlocking screens

### DISCUSSION THREAD

**Discussion by rdonkin** [MonitorControl/MonitorControl#1502] [General]
### Before opening the issue, have you...?

- [X] Searched for existing issues
- [X] Looked through [the wiki](https://github.com/MonitorControl/MonitorControl/wiki)
- [X] Updated MonitorControl to the latest version (if applicable)

### Describe the bug

When MonitorControl is running, this can happen:

1. Lock screen via Apple menu (going to screensaver mode, not a monitor sleep or system sleep)
2. Unlock  by logging in

Symptom is that all windows on monitor are moved to the Macbook display. 

Same happens if monitor is disconnected (USB-C cable) and reconnected.

This is only _somewhat_ reproducible - but it has only been happening since I installed MonitorControl recently.

Workaround: kill MonitorControl (had been running for 22 days), and the above does not happen.

- it may be enough to restart MonitorControl - after one restart, problem didn't happen

### Steps to reproduce

1. Put some windows on external display while MonitorControl is running
2. Apple menu, Lock Screen
3. Log in to unlock
4. Windows will have moved from external display to Macbook display

Can also reproduce by disconnecting and re-connecting monitor (with USB-C cable).

Workaround:
1. Kill MonitorControl process
2. Repeat above sequence - windows will not move

### Expected behavior

Windows don't move between displays after locking screen.

### Anything else?

Macbook display is set as main display in Display system preferences.  Monitor is set as extended display.

Mission Control preferences: set to "displays have separate Spaces"

If Macbook is put into Sleep mode, the windows do move even if MonitorControl is not running.

### Environment Information (please complete the following information)

```markdown
- macOS version: 12.7.1
- Mac model: Macbook Air M1 2020
- MonitorControl version: v4.2.0
- Monitor(s): Philips 272B7QUB
- Apple Silicon/M1 (yes or no): yes
```

**Comment by rdonkin** [comment]
After a reboot due to macOS 12.7.2 update, this hasn't yet recurred - will report back if it does.

I really like MonitorControl btw so I will try to restart the process if this does recur.

**Comment by MPFGlaser** [comment]
Interestingly enough, I've been experiencing similar behaviour over the past months. Only caveat being that I installed MonitorControl only today, so it might be on the MacOS side of things.

**Comment by rdonkin** [comment]
Thanks for the comment @MPFGlaser.   It seems there are many other causes for this behaviour, even though it appears that MonitorControl may be involved in my case.

Reports of similar issues where MonitorControl is not mentioned, with various triggers:

- https://apple.stackexchange.com/questions/401432/how-to-stop-mac-from-moving-windows-from-external-display-to-laptop-when-opening
- https://discussions.apple.com/thread/253787680?sortBy=best - many similar threads here

So your issue may not be related to MonitorControl.  Next time it happens, maybe see if killing or restarting MonitorControl solves this?

**Comment by waydabber** [comment]
This is most probably not a MonitorControl issue. You usually need to reset the windowserver configuration which manages display layouts. To do that, do steps 3,4 and 9 in this tutorial: https://github.com/waydabber/BetterDisplay/wiki/Safe-mode,-app-reset,-app-removal#exhaustive-reset
